File: Cc.Core/Models/Event.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models
{
    public partial class Event
    {
        public Event()
        {
            EventAttachment = new HashSet<EventAttachment>();
        }

        public long Id { get; set; }
        public long? OrderId { get; set; }
        public bool Closed { get; set; }
        public bool? LawyerVisible { get; set; }
        public string? Description { get; set; }
        public DateTime? EventDate { get; set; }
        public string? AttachmentName { get; set; }
        public long? EventTypeId { get; set; }
        public string? InsertUser { get; set; }
        public DateTime? InsertDate { get; set; }
        public string? UpdateUser { get; set; }
        public DateTime? UpdateDate { get; set; }

        public virtual EventType? EventType { get; set; }
        public virtual OrderTable? Order { get; set; }
        public virtual ICollection<EventAttachment> EventAttachment { get; set; }
    }
}

File: Cc.Ui/src/models/event.ts
import { Entity, resource, field, required } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';

@resource('event')
export class Event extends Entity {

    @field()
    id: number;
    @field()
    orderId: number;
    @field()
    closed: boolean;
    @field()
    lawyerVisible: boolean;
    @field()
    description: string;
    @field()
    eventDate: string;
    @field()
    eventTypeId: number;
    @field()
    insertUser: string;
    @field()
    insertDate: string;
    @field()
    updateUser: string;
    @field()
    updateDate: string;

    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): Event {

        let item = new Event();
        return item;
    }
}

@resource('event')
export class EventSearchModel extends Entity implements Pagination {
    
    @field()
    orderId?: number;
    @field()
    closed?: boolean;
    @field()
    lawyerVisible?: boolean;
    @field()
    description?: string;
    @field()
    eventDate?: string;
    @field()
    eventTypeId?: number;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}